packages/astro/src/@types/astro.ts

```diff
@@ -899,8 +901,29 @@ export interface AstroUserConfig {
 		 *   }
 		 * }
 		 * ```
+		 *
+		 * If this value is a `Record<string, string>`,
+		 *
+		 * Applicable to multiple CDN scenarios. If you static file use different CDN, you can configure it like the example below.
+		 *
+		 * For example, if static file is js file, assets prefix use `https://js.cdn.example.com`. If static file is css file, assets prefix use `https://css.cdn.example.com`. Other assets prefix use `https://cdn.example.com`.
+		 *
+		 * ```js
+		 * {
+		 *   build: {
+		 *     assetsPrefix: {
+		 *       'js': 'https://js.cdn.example.com',
+		 *       'css': 'https://js.cdn.example.com',
+		 *       'fallabck': 'https://cdn.example.com'
+		 *     }
+		 *   }
+		 * }
+		 * ```
+		 *
+		 * Object key is file extension, value is CDN assets prefix path. `fallabck` is default assets prefix cdn value set and it is a mandatory key. You can set the default values of other files by setting the value of `fallabck`.
```

Comment:
```suggestion
		 * If an object of `Record<string, string>` is passed, you can individually map certain file extensions to use a specific assets prefix. This can be useful for multiple CDN scenarios, for example:
		 *
		 * ```js
		 * {
		 *   build: {
		 *     assetsPrefix: {
		 *       js: 'https://js.cdn.example.com',
		 *       mjs: 'https://js.cdn.example.com',
		 *       css: 'https://css.cdn.example.com',
		 *       fallback: 'https://cdn.example.com'
		 *     }
		 *   }
		 * }
		 * ```
		 *
		 * The `fallback` key must be set to so unspecified file extensions will fallback to this value.
```

I think this jsdoc will be slightly clearer and concise.

Reply:
Also resolving this in favour of Sarah's review below